src/udx.c

```diff
@@ -1888,6 +1901,59 @@ udx_stream_check_timeouts (udx_stream_t *handle) {
   return err < 0 ? err : 0;
 }
 
+int
+udx_stream_change_remote (udx_stream_t *stream, uint32_t remote_id, const struct sockaddr *remote_addr, udx_stream_remote_changed_cb on_remote_changed) {
+  assert(stream->status & UDX_STREAM_CONNECTED);
+  if (!(stream->status & UDX_STREAM_CONNECTED)) {
+    return UV_EINVAL;
+  }
+
+  if (remote_addr->sa_family == AF_INET) {
+    stream->remote_addr_len = sizeof(struct sockaddr_in);
+    if (((struct sockaddr_in *) remote_addr)->sin_port == 0) {
+      return UV_EINVAL;
+    }
+  } else if (remote_addr->sa_family == AF_INET6) {
+    stream->remote_addr_len = sizeof(struct sockaddr_in6);
+    if (((struct sockaddr_in6 *) remote_addr)->sin6_port == 0) {
+      return UV_EINVAL;
+    }
+  } else {
+    return UV_EINVAL;
+  }
+
+  memcpy(&stream->remote_addr, remote_addr, stream->remote_addr_len);
+  stream->remote_id = remote_id;
+
+  printf("stream->seq=%d stream->remote_acked=%d\n", stream->seq, stream->remote_acked);
+
+  if (stream->seq != stream->remote_acked) {
+
+    stream->remote_changing = true;
+    stream->seq_on_remote_changed = stream->seq;
+    stream->on_remote_changed = on_remote_changed;
+  } else {
+
+    printf("ready to change now\n");
+    on_remote_changed(stream);
+  }
+
+  stream->mtu = UDX_MTU_BASE;
+  stream->mtu_state = UDX_MTU_STATE_BASE;
+  stream->mtu_probe_count = 0;
+  stream->mtu_probe_size = UDX_MTU_BASE; // starts with first ack, counts as a confirmation of base
+  stream->mtu_max = UDX_MTU_MAX;         // revised in connect()
+
+  uv_timer_stop(&stream->mtu_raise_timer);
+
+  if (stream->socket->family == 6 && stream->remote_addr.ss_family == AF_INET) {
+    addr_to_v6((struct sockaddr_in *) &stream->remote_addr);
+    stream->remote_addr_len = sizeof(struct sockaddr_in6);
+  }
```

Comment:
Should this be moved to _before_ the `on_remote_changed` callback is invoked just to be safe?

Reply:
suggestion taken